pegasus/helpers/email_preference_helpers.rb

```diff
@@ -47,5 +37,21 @@ def self.upsert!(email:, opt_in:, ip_address:, source:, form_kind:)
           updated_at: current_time
         }
       )
+    # Don't change opt_in to false if a preference with opt_in = true exists already.
+    # We currently only enable a user to opt out via the Pardot unsubscribe link.
+    elsif !existing_email_preference[:opt_in] || input_opt_in
+      Dashboard.db[:email_preferences].update(
+        {
+          opt_in: input_opt_in,
+          source: source,
+          ip_address: ip_address,
+          form_kind: form_kind,
+          # Don't set created_at on an update.
+          updated_at: current_time
+        }
+      )
+    end
+    # This is a write-only helper method.  Don't return data to the caller.
+    return
```

Comment:
Again, why do we need this? Is there harm in allowing the (unused) return? I don't think we want to start adding nil returns to every Ruby function, and besides it's against the [style guide](https://github.com/bbatsov/ruby-style-guide#no-explicit-return), at least in this form. My interpretation of that is we'd prefer `nil` over `return`, but again I don't think we need either in this case.